lib/moneta/transformer.rb

```diff
@@ -77,13 +80,26 @@ def without(*options)
         options.empty? ? 'options' : "Utils.without(options, #{options.map(&:to_sym).map(&:inspect).join(', ')})"
       end
 
-      def compile_key_transformer(klass, key, key_opts)
+      def compile_each_key_support_clause(klass, keys)
         klass.class_eval <<-END_EVAL, __FILE__, __LINE__ + 1
-          not_supports :each_key
+          #{'not_supports :each_key' if @load_key_validator !~ keys.map(&:inspect).join}
+        END_EVAL
+      end
 
+      def compile_key_transformer(klass, key, key_opts, key_load, key_load_opts)
+        klass.class_eval <<-END_EVAL, __FILE__, __LINE__ + 1
           def key?(key, options = {})
             @adapter.key?(#{key}, #{without key_opts})
           end
+          def each_key(&block)
+            raise NotImplementedError, "each_key is not supported on this transformer" \
+              unless supports? :each_key
+
+            return enum_for(:each_key) { @adapter.each_key.size } unless block_given?
+            @adapter.each_key.lazy.map{ |key| #{key_load} }.each(&block)
```

Comment:
Would `@adapter.each_key{ |key| yield #{key_load} }` work here?

Reply:
Just thinking about this more:  say for example, you are using a Redis store with the `prefix` key transformer.  If you call `each_key`, it looks like you will get all the keys in the store, whether they are prefixed or not.  I don't think the behaviour of `each_key` in this situation is super-well defined, but I think it's probably at least arguable that a user would expect to only get prefixed keys - what do you think?

Reply:
I guess you are right, but for now it fits my use case. We could either filter it in weird way inhere or let them know that they have to filter these themselves?

We should probably have a custom block that filters and yields only the ones prefixed?